src/encauth/chachapoly/chacha20poly1305_done.c

```diff
@@ -27,7 +27,7 @@ int chacha20poly1305_done(chacha20poly1305_state *st, unsigned char *tag, unsign
 
    LTC_ARGCHK(st != NULL);
 
-   padlen = 16 - (st->ctlen % 16);
+   padlen = 16 - (unsigned long)(st->ctlen % 16);
```

Comment:
Here it is a bit complicated as `ctlen` must be 64bit (at least) whereas padlen is simply a type we use for buffer length (today `unsigned long`, maybe `size_t` in the future).

Making `ctlen` smaller than 64bit (which unsigned long is for example on MS Windows/x64) will limit the max length of data that can be encrypted via `chacha20poly1305`.

Now when I am looking at the implementation the max length of our `chacha20poly1305` is `2^64` bytes; however, in theory `chacha20poly1305` has max length limit `2^70` bytes. The potentioal trouble might occur due to the fact that we do not check for reaching `2^64`. This is perhaps a bit OT here, anyway `ctlen` must be 64bit.